docs/dev/RELEASE.md

```diff
@@ -0,0 +1,87 @@
+## Introduction
+This document guides maintainers through the process of creating an Envoy Gateway release.
+
+## Prerequisites
+- Permissions to push to the Envoy Gateway repository.
+
+## Creating a Minor Release
+
+1. Clone the repo, checkout the `main` branch, ensure it’s up-to-date, and your local branch is clean.
+2. Create a changelog corresponding to the release number that summarizes the changes included in the release.
+   Reference previous [changelogs][changelogs] for additional details.
+3. Submit a [Pull Request][pr] to merge the changelog into the main branch. This should be the last commit to main
+   before cutting the release.
+4. Create a new release branch from `main`. The release branch should be named
```

Comment:
do we need 4. atm ?

Reply:
I believe we need a release branch for minor releases so we can a) backport bug fixes b) push PRs for release-specific changes, e.g. quickstart guide.